### PR TITLE
Migrate freeze-test to cpp (RIPD-1154)

### PR DESCRIPTION
**Pull request by mellery451**
Port all active tests in freeze-test.coffee to c++.

### CODE REVIEW

File: src/test/app/Freeze_test.cpp [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+
+class Freeze_test : public beast::unit_test::suite
+{
+
+    static Json::Value
+    getAccountLines(test::jtx::Env& env, test::jtx::Account const& account)
+    {
+        Json::Value jq;
+        jq[jss::account] = account.human();
+        return env.rpc("json", "account_lines", to_string(jq))[jss::result];
```
Comment (miguelportilla) [inline src/test/app/Freeze_test.cpp]: Another option is boost::lexical_cast<std::string>(jq)

File: src/test/app/Freeze_test.cpp [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+        env(pay(G1, bob, G1["USD"](10)));
+        env(pay(G1, alice, G1["USD"](100)));
+        env.close();
+
+        env(offer(alice, XRP(500), G1["USD"](100)));
+        env.close();
+
+        {
+            auto lines = getAccountLines(env, bob);
+            BEAST_EXPECT(checkArraySize(lines[jss::lines], 1u));
```
Comment (miguelportilla) [inline src/test/app/Freeze_test.cpp]: Consider returning if checkArraySize returns false.

File: src/test/app/Freeze_test.cpp [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+
+        {
+            // check G1 account lines
+            //    test: shows freeze
+            auto lines = getAccountLines(env, G1);
+            Json::Value bobLine;
+            for (auto const& it : lines[jss::lines])
+            {
+                if (it[jss::account] == bob.human())
+                    bobLine = it;
```
Comment (miguelportilla) [inline src/test/app/Freeze_test.cpp]: Suggestion:
`
bobLine = it;
break;
`

File: src/test/app/Freeze_test.cpp [hunk 4]
```diff
@@ -0,0 +1,520 @@
...
+        }
+
+        {
+            //    test: shows freeze peer
+            auto lines = getAccountLines(env, bob);
+            Json::Value g1Line;
+            for (auto const& it : lines[jss::lines])
+            {
+                if (it[jss::account] == G1.human())
+                    g1Line = it;
```
Comment (miguelportilla) [inline src/test/app/Freeze_test.cpp]: Suggestion:
`g1Line = it; break;`

**Comment by miguelportilla** [comment]
Left some minor comments. Test looks great, nice job @mellery451
👍

File: src/test/app/Freeze_test.cpp (hunk 2 above)
Comment (mellery451) [inline src/test/app/Freeze_test.cpp]: done (multiple locations)

File: src/test/app/Freeze_test.cpp (hunk 3 above)
Comment (mellery451) [inline src/test/app/Freeze_test.cpp]: done - thx

**Comment by nbougalis** [comment]
Merged as 28ae522ea2d85d066ceeb23d30ade451fd1e424d.

**Pull request closed without merging**